pkg/controller/azvolume.go

```diff
@@ -140,7 +140,26 @@ func (r *reconcileAzVolume) TriggerDelete(ctx context.Context, volumeName string
 	if err := r.UpdateStatus(ctx, azVolume.Name, true); err != nil {
 		return err
 	}
-	klog.Infof("successfully deleted volume (%s)and update status of AzVolume (%s)", azVolume.Spec.UnderlyingVolume, azVolume.Name)
+
+	// Delete all AzVolumeAttachment objects bound to the deleted AzVolume
+	attachments, err := r.azVolumeClient.DiskV1alpha1().AzVolumeAttachments(r.namespace).List(ctx, metav1.ListOptions{})
```

Comment:
As I mentioned in the meeting this morning, it is not expected that the `AzVolume` be deleted while still in use (and in that case attached. There should be no `AzVolumeAttachment` instances in this case. Indeed, if the underlying volume is still attached to one or more nodes, the call to `DeleteAzVolume` above should fail.

I suppose we could consider this opportunistic clean up of leaked `AzVolumeAttachment` instances and its controller would need to gracefully handle the case where the managed disk is no longer attached to the expected node.